server/src/main/java/org/elasticsearch/index/search/QueryStringQueryParser.java

```diff
@@ -467,11 +467,11 @@ protected Query getFuzzyQuery(String field, String termStr, float minSimilarity)
 
     private Query getFuzzyQuerySingle(String field, String termStr, int minSimilarity) throws ParseException {
         MappedFieldType currentFieldType = context.getFieldType(field);
-        if (currentFieldType == null) {
+        if (currentFieldType == null || currentFieldType.getTextSearchInfo() == TextSearchInfo.NONE) {
```

Comment:
QueryStringQueryParser tries not to throw exceptions; the idea is that can chuck any old rubbish in there and it will attempt to build a valid query from it.

Reply:
right, that makes sense. so without this change, it would break because there is no analyzer set if you query the wrong field?

Reply:
yes

Reply:
would it make sense to have a test for this? I see we are a bit light on testing for query string, I think. Maybe we should address that as a followup?

Reply:
Most of the testing is in QueryStringQueryBuilderTests I think, and it does look like we have no tests for completely unsearchable fields - let's add in a followup?